crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -6,35 +6,46 @@ use mqtt_channel::StreamExt;
 use mqtt_channel::Topic;
 use mqtt_channel::TopicFilter;
 use reqwest::Url;
+use std::collections::HashMap;
 use std::time::Duration;
 use tedge_config::mqtt_config::MqttConfigBuildError;
 use tedge_config::TEdgeConfig;
 use tracing::error;
 use tracing::info;
 
+#[derive(thiserror::Error, Debug)]
+pub enum C8yEndPointError {
+    #[error("Internal id is empty")]
+    EmptyInternalID,
+}
+
 /// Define a C8y endpoint
 #[derive(Debug)]
 pub struct C8yEndPoint {
     pub c8y_host: String,
     pub device_id: String,
-    pub c8y_internal_id: String,
+    pub token: Option<String>,
+    pub devices_internal_id: HashMap<String, String>,
```

Comment:
The asymmetry of this struct makes things fragile and confusing.

The asymmetry is that the main device id is stored in in its own field while the internal-id is stored in a map along all the child devices' ids and external ids.

Reply:
Agreed, now refactored the map to hold `DeviceType` enum and `internal-id`.